packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -329,12 +329,24 @@ class IconButton extends StatelessWidget {
   Widget build(BuildContext context) {
     assert(debugCheckHasMaterial(context));
     final ThemeData theme = Theme.of(context);
+
+    // Ancestor IconThemes could be found in the direct ancestor path, or, if
+    // this IconButton appears in an AppBar, then in that AppBar's `iconTheme`
+    // attribute.
+    final AppBar? appBar = context.findAncestorWidgetOfExactType<AppBar>();
+    final IconThemeData iconTheme = appBar?.actionsIconTheme ?? appBar?.iconTheme ?? theme.iconTheme;
```

Comment:
Instead of hard-coding in the appBar, shouldn't this just look up the IconTheme injected by the appbar via IconTheme.of(context)?

Reply:
Oh my goodness, I didn't realize AppBar inserted that into the main theme to make it discoverable in this way. That works!